charts/tekton-pipelines/templates/buildpacks/pipelines/_buildpack.yaml

```diff
@@ -161,6 +168,12 @@ spec:
       runAfter:
         - kustomize
 
-  {{ include "task.finalNotification" . | nindent 2 }}
+  {{- include "task.finalNotification" . | nindent 2 }}
+  {{- end }}
+
+  {{- if .buildOnly }}
```

Comment:
Can you leave this `include "task.finalNotification"` as it was without if else condition? you can define what should be passed to this include with separate variable. So no need to copy this block in separate `if` conditions with different params. You can also use `ternary` for that:
```
{{- $env := ternary .Values.localEnv .Values.prodEnv .Values.isLocal -}}
```